pkg/cloudencrypt/encode.go

```diff
@@ -0,0 +1,59 @@
+package cloudencrypt
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/base64"
+	"encoding/gob"
+
+	"github.com/pkg/errors"
+)
+
+func encode(data any) ([]byte, error) {
```

Comment:
I would move this under `internal` package
https://github.com/prometheus/client_golang/tree/main/prometheus
This is prometheus library, it has all the exported interface included in top level package (in our case `/pkg`) the implementation details are stored in internal, because this is something that won't be any time soon exported within `/pkg`. Also you want to prevent including this specific encoding and decoding by other users. That's the intent of having `internal` modules

Reply:
I don't like that approach to be honest. With that package you need to make those functions exported which means someone else can import and use them which is unintended. I don't see a reason for that approach when Go language has the native feature of only exporting capitalized functions.

Reply:
Have you tried import something from `internal` from a library? I think that's what you miss, it is not possible. When you have our `go-client` library, basically, everyting that you would like to import into your project and lays in `github.com/keboola/go-client/internal` is impossible. That's how the module system works in go.
Yea, go allows you to do both, export within `internal` module and export only within single `any` module. 

The reason behind this is simply that someone would come and do PR in which is changes the name to `Encode(data any) ([]byte, error)` and it is exported and part of library. But in this case it fits only our purpose and it is not general `Encode()` function.
Again, I will leave that but I mind that maybe you did not try that, what I suggest. Maybe you will it find out usefull. 
https://github.com/golang-standards/project-layout

Reply:
Alright, yeah this is an important information which I was missing. But I'm having trouble finding a documentation about this topic. Note that the repository you linked is no standard, just someone presenting their preferred layout under misleading organization name. I'm not refusing using the `internal` packages but I need a better source than this extremely controversial repository.

Reply:
https://go.dev/doc/modules/layout#package-or-command-with-supporting-packages

Reply:
Alright, now we're getting somewhere. However I still don't think it's a best practice to move every single non-exported function into an internal package. I checked other open source go libraries and even those that are using internal packages still have non-exported functions outside of the `internal` directory.

I agree that `encode.go` should be moved into an internal package. I'll have to double check the other non-exported functions.

Reply:
Here, I moved what made sense to me: https://github.com/keboola/go-cloud-encrypt/commit/bcb2cf2432763b2233ab60eb0b2ea78605282306

However it doesn't make sense to me to move these: https://github.com/keboola/go-cloud-encrypt/blob/bcb2cf2432763b2233ab60eb0b2ea78605282306/pkg/cloudencrypt/dual.go#L81-L113

Reply:
I agree, this is something that native encryptor uses but that things were generic/ used accross and it is quite out implementation of encoding. If we would like to be proper, the encoder could be interface but I am fine with this :+1: